### PR TITLE
React index page

### LINKED ISSUE #429: switch over course index page
Might just keep the existing styling as currently done. 
Screen Shot 2019-03-12 at 7 44 07 AM

### PR DESCRIPTION
**Pull request by Adonais0**
closes #429

### CODE REVIEW

File: assets/src/containers/IndexPage.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+      },
+    wrapper: {
+        marginTop: theme.spacing.unit * 2,
+        flexDirection: "row",
+        justifyContent: "center"
+    },
+})
+
+function IndexPage (props) {
+    const { classes, location } = props;
```
Comment (justin0022) [inline assets/src/containers/IndexPage.js]: Could you change this to:
```javascript
const { classes, match } = props
const currentCourseId = match.params.courseId
```
The `match` property explicitly sends down the `courseId` param from `App.js`, making the implementation more robust.

File: assets/src/containers/IndexPage.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            <Grid item xs={12}>
+                <Grid 
+                container 
+                className={classes.wrapper} 
+                spacing={8}
+                >
+                {
+                    routes(courseId).map((props, key) => 
+                    <Link style={{ textDecoration: 'none' }} to={props.path} key={key}>
+                        <SelectCard cardData = {props} images={images} />
```
[the remark this answers is not in the document]
Comment (Adonais0) [reply assets/src/containers/IndexPage.js]: sorry about that, forgot to remove after testing

File: assets/src/containers/IndexPage.js (hunk 1 above)
Comment (Adonais0) [reply assets/src/containers/IndexPage.js]: sure, done

File: assets/src/containers/IndexPage.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (justin0022) [reply assets/src/containers/IndexPage.js]: No worries

**Comment by pushyamig** [comment]
I get below error in  browser console and index page is not at all loading. 
`Failed context type: The context `router` is marked as required in `Link`, but its value is "undefined"`

**Pull request merged**